File: go/longest-common-prefix/longest-common-prefix.go
package main

import(
	"fmt"
	"sort"
)

func longestCommonPrefix(strs []string) string {
    if len(strs) == 0 {
        return ""
    }
    sort.Strings(strs)
    first, last := strs[0], strs[len(strs)-1]
    result := ""
    for i := 0; i < len(first); i++ {
        if i < len(last) && first[i] == last[i] {
            result += string(first[i])
        } else {
            break
        }
    }
    return result
}

func main(){
	array := []string{"flow", "flower", "flight"}
	fmt.Println("answer to the first example is:",longestCommonPrefix(array))
}

File: js/longest-common-prefix/longest-common-prefix.js
function longestCommonPrefix(strs){
    if (strs.length == 0) {
        return ""
    }
    strs.sort()
    let first = strs[0], last = strs[strs.length-1]
    let result = ""
    for (i = 0 ; i < first.length ; i++){
        if (i < last.length && first[i] == last[i]){
            result += first[i].toString()
        }else{
            break
        }
    }
    return result
}

let array = ["flow", "flower", "flight"]
console.log("answer to the first example is:",longestCommonPrefix(array))